blaze-client/src/main/scala/org/http4s/blaze/client/Http1Connection.scala

```diff
@@ -317,98 +316,106 @@ private final class Http1Connection[F[_]](
       cb: Callback[Response[F]],
       idleTimeoutS: F[Either[Throwable, Unit]],
   ): Unit =
-    try if (!parser.finishedResponseLine(buffer))
-      readAndParsePrelude(cb, closeOnFinish, doesntHaveBody, "Response Line Parsing", idleTimeoutS)
-    else if (!parser.finishedHeaders(buffer))
-      readAndParsePrelude(cb, closeOnFinish, doesntHaveBody, "Header Parsing", idleTimeoutS)
-    else {
-      // Get headers and determine if we need to close
-      val headers: Headers = parser.getHeaders()
-      val status: Status = parser.getStatus()
-      val httpVersion: HttpVersion = parser.getHttpVersion()
-
-      // we are now to the body
-      def terminationCondition(): Either[Throwable, Option[Chunk[Byte]]] =
-        stageState.get match { // if we don't have a length, EOF signals the end of the body.
-          case Error(e) if e != EOF => Either.left(e)
-          case _ =>
-            if (parser.definedContentLength() || parser.isChunked())
-              Either.left(InvalidBodyException("Received premature EOF."))
-            else Either.right(None)
-        }
+    try
```

Comment:
wow, that method definitely should be refactored 😮

Reply:
well, someone had to do it #5789

Reply:
We should probably merge that and let steward rebase afterward?

Reply:
@rossabaker we can either merge this and then I'll rebase. Or vice-versa. I'd appreciate any scenario.